src/primitive_core.cc

```diff
@@ -1814,18 +1819,20 @@ PRIMITIVE(process_send) {
 
   SystemMessage* message = null;
   MessageEncoder encoder(process, buffer);
-  bool encoding_succeeded = false;
-  if (encoder.encode(array)) {
-    encoding_succeeded = true;
-    message = _new SystemMessage(type, process->group()->id(), process->id(), buffer);
+  if (!encoder.encode(array)) {
+    encoder.free_copied();
```

Comment:
Maybe we can flip it around so you have to call something on the encoder to get it to not free the copies and the incoming buffer? That way we might be able to get away with just returning here.

It is essentially the `neuter_externals` call, I think.